pkg/kms/localkms/localkms.go

```diff
@@ -0,0 +1,195 @@
+/*
+ Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package localkms
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/core/registry"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/signature"
+	tinkpb "github.com/google/tink/proto/tink_go_proto"
+
+	"github.com/hyperledger/aries-framework-go/pkg/kms/localkms/internal/tinkwrapper"
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	nameSpace = "kmsdb"
+)
+
+type provider interface {
+	StorageProvider() storage.Provider
+	SecretLock() secretlock.Service
+}
+
+// LocalKMS implements kms.KeyManager to provide key management capabilities using a local db.
+// It uses an underlying secret lock service (default local secretLock) to wrap (encrypt) keys
+// prior to storing them.
+type LocalKMS struct {
+	secretLock       secretlock.Service
+	masterKeyURI     string
+	store            storage.Store
+	masterKeyEnvAEAD *aead.KMSEnvelopeAEAD
+}
+
+// NewKMS will will create a new KMS service
+func NewKMS(masterKeyURI string, p provider) (*LocalKMS, error) {
+	s, err := p.StorageProvider().OpenStore(nameSpace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to ceate local kms: %w", err)
+	}
+
+	secretLock := p.SecretLock()
+
+	kmsClient, err := tinkwrapper.NewClient(secretLock, masterKeyURI)
+	if err != nil {
+		return nil, fmt.Errorf("failed to ceate local kms client: %w", err)
+	}
+
+	// register kmsClient in Tink
+	registry.RegisterKMSClient(kmsClient)
```

Comment:
It seems weird that we are hardcoding registration in tink as part of a constructor.

Reply:
suggestions?

`registry.RegisterKMSClient(kmsClient)` can be moved out of the constructor and called after the creation of LocalKMS.. I will have to expose kmsClient though.

Reply:
I was also going to ask why we actually need to use Tink's singleton instead of keeping an internal reference to the KeyManager where needed?

Reply:
I've commented out Tink's singleton call. It's mainly used for cloud KMSs, so its not needed for localKMS

Reply:
after reviewing this commented out code, I realized that LocalKMS does not need a KMS client to connect to, it only needs a key wrapper represented as `masterKeyEnvAEAD`

I refactored the code to remove KMSClient and added a New constructor in LocalAEAD to prepare the key wrapper for LocalKMS. 

For KMS implementations other than LocalKMS and are hosted on the cloud, Tink offers a KMS client and AEAD wrapper for the major cloud providers, ie AWS KMS, GCP KMS, Hashcorp KMS.

With this last patch, the code is cleaner this way.